analytics-core/src/main/scala/org/ekstep/analytics/framework/util/MergeUtil.scala

```diff
@@ -74,8 +74,29 @@ class MergeUtil {
           columnOrder = columnOrder.filter(col=> reportDfColumns.contains(col))
           MergeResult(mergeReport(rollupCol,rollupFormat,deltaDF, reportDF, mergeConfig, mergeConfig.merge.dims), reportDF,
             StorageConfig(storageType, postContainer, path.getParent.getName,Option(storageKey),Option(storageSecret)))
+        case "s3" =>
+          var deltaDF = fetchOSSFile(filePaths("deltaPath"),
```

Comment:
What is the use of this deltaDF?

Reply:
This block of code for case s3 , is a direct replica for the case of azure, with the relevant difference like 
var deltaDF = fetchBlobFile(filePaths("deltaPath") changed to var deltaDF = fetchOSSFile(filePaths("deltaPath")

here s3 refers to the deep storage type used by druid. There was no implementation when the deep storage used by druid is of type s3.
This block of change was to fill the gap for s3 storage type with the existing business logic of type azure.